### PR TITLE
Improve simulator functionality

### PR DESCRIPTION
**Pull request by Modularius**
## Summary of changes

1. Changed `unwrap`s to `expect`s in `build_messages.rs`
2. Changed `send_trace_message` to `send_digitiser_trace_message` and integrated edits in `send_messages.rs`,  `simulation.rs`, `engine.rs` and others.
3. Added `Trace` struct for channel traces in `event_lists.rs` and refactored
4. Added `IntConstant` and `TextConstant` types
5. Added `EnsureDelayMs` action
6. Improved error handling

## Instruction for review/testing

General code review as the simulator was successfully tested in the pipeline.

### CODE REVIEW

**Review by DanNixon — CHANGES_REQUESTED** [review]
I don't like the amount of `unwrap` and `expect`. Now that we have a standard for how error handling should happen I would like new and refactored code to use it (arguably it is of less concern for the simulator but I see a lot of the same elsewhere, anything related to tracing for example).

File: simulator/src/integrated/simulation_elements/utils.rs [hunk 1]
```diff
@@ -26,6 +26,40 @@ impl FloatExpression {
...
+    Int(i32),
+    IntEnv(String),
+}
+
+impl IntConstant {
+    pub(crate) fn value(&self) -> i32 {
+        match self {
+            IntConstant::Int(v) => *v,
+            IntConstant::IntEnv(environment_variable) => {
+                env::var(environment_variable).unwrap().parse().unwrap()
```
[the remark this answers is not in the document]
Comment (Modularius) [reply simulator/src/integrated/simulation_elements/utils.rs:41]: It is useful for varying the number of digitisers. As changing this requires the aggregator to be restarted, the only safe way to do it is to restart the whole pipeline with the number of digitisers defined by environment variable.

**Comment by Modularius** [comment]
There were considerably more `unwrap`s than I realised. I've replaced them with `anyhow`s (they inform the user of invalid JSON), and all remaining `except`s should be impossible.

**Comment by DanNixon** [comment]
> I've replaced them with `anyhow`s (they inform the user of invalid JSON)

See https://github.com/STFC-ICD-Research-and-Design/supermusr-data-pipeline/blob/main/docs/style.md#crab-error-handling-rules, `anyhow` outside of the return type of `main()` is prohibited.

**Review by DanNixon — APPROVED** [review]

**Pull request merged**